test-records.js

```diff
@@ -77,26 +94,75 @@ validator('companies', cdb_schema, (json) => {
-                        'See: https://github.com/datenanfragen/data/issues/1120'
-                    );
+                    errors.push({
+                        msg: `Record specifies '${prop}' of '${json[prop]}' but that isn't available in English.See: https://github.com/datenanfragen/data/issues/1120`,
+                    });
             }
         }
     }
 
     // A `quality` of `tested` may only be set if `required-elements` are specified (#811).
     if (json['quality'] === 'tested') {
         if (!json['required-elements'])
-            fail(
-                "Record has `quality` of `tested` but doesn't specify `required-elements`.",
-                'See: https://github.com/datenanfragen/data/issues/811'
+            errors.push({
+                msg: "Record has `quality` of `tested` but doesn't specify `required-elements`. See: https://github.com/datenanfragen/data/issues/811",
+            });
+    }
+    // whitespace check
+    Object.keys(json).forEach((key) => {
+        if (typeof json[key] === 'string' && json[key] !== json[key].trim())
+            errors.push(`Seems like \`${key}\` isn't trimmed, i.e. it contains leading or trailing whitespace.`);
+    });
+
+    //// ADDRESS FORMATTING
+    const address_lines = json['address'].split('\n');
+    if (address_lines.length < 2) errors.push('`address` is not formatted with newlines (\\n).');
+    for (line of address_lines) {
+        if (line !== line.trim()) {
+            ("`address` isn't trimmed, i.e. contains unnecessary whitespace.");
+        }
+    }
+    if (address_lines[0].trim() === json['name']) errors.push(' Record includes `name` in the `address`.');
+
+    // check if the last line is a country
+    // this test might produce false-positives, as it's a difficult thing to do
+    // we might consider more fuzzy matching if we get many reports of false positives
+    let last_line_is_country = false;
+    const last_line = address_lines[address_lines.length - 1].trim();
+    for (const [countrycode, v] of Object.entries(countries)) {
+        if ([countrycode, v.name, v.native].includes(last_line)) {
+            last_line_is_country = true;
+            break;
+        }
+        // check for substring match (Netherlands vs *The *Netherlands)
+        if (
+            [countrycode, v.name, v.native].some((x) => {
+                return last_line.includes(x) || x.includes(last_line);
+            })
+        ) {
+            last_line_is_country = true;
+        }
+    }
+    // TODO: replace England with United Kingdom in the DB
+    if (['United States of America', 'England']) last_line_is_country = true;
+    if (!last_line_is_country)
+        errors.push(
+            'Last line of `address` should be a country. If you feel like this error is a mistake, please let us now! We get our list of countries from https://www.npmjs.com/package/countries-list'
+        );
+
+    // set suggested-transport-medium if email is privacy related
```

Comment:
Oh, and speaking of hard fails: This cannot be one. :D This can only be a suggestion (which won't work through this framework) as there may well be companies with an email matching the regex but where we would suggest a different transport medium.